go-controller/pkg/ovn/policy.go

```diff
@@ -449,31 +442,36 @@ func (oc *Controller) updateACLLoggingForPolicy(np *networkPolicy, logLevel stri
 	if !np.created {
 		return NetworkPolicyNotCreated
 	}
-
-	acls := oc.buildNetworkPolicyACLs(np, logLevel)
-	ops, err := libovsdbops.UpdateACLsLoggingOps(oc.nbClient, nil, acls...)
-	if err != nil {
-		return err
+	// Predicate for given network policy ACLs
+	p := func(item *nbdb.ACL) bool {
+		return item.ExternalIDs[namespaceACLExtIdKey] == np.namespace && item.ExternalIDs[policyACLExtIdKey] == np.name
 	}
-	_, err = libovsdbops.TransactAndCheck(oc.nbClient, ops)
-	return err
+	return UpdateACLLoggingWithPredicate(oc.nbClient, p, aclLogging)
 }
 
 func (oc *Controller) setNetworkPolicyACLLoggingForNamespace(ns string, nsInfo *namespaceInfo) error {
-	var ovsDBOps []ovsdb.Operation
-	for _, policyType := range []knet.PolicyType{knet.PolicyTypeIngress, knet.PolicyTypeEgress} {
-		denyACL, _ := buildDenyACLs(ns, "", targetPortGroupName(nsInfo.portGroupIngressDenyName, nsInfo.portGroupEgressDenyName, policyType), nsInfo.aclLogging.Deny, policyType)
-		var err error
-		ovsDBOps, err = libovsdbops.UpdateACLsLoggingOps(oc.nbClient, ovsDBOps, denyACL)
-		if err != nil {
-			return err
-		}
+	// update namespace-scoped default deny ACLs
+	// default network policy ACLs don't have any special name or externalIDs,
+	// therefore we can get all ACLs referenced by defaultDeny port groups
+	defaultDenyPGPred := func(item *nbdb.PortGroup) bool {
+		return item.Name == nsInfo.portGroupEgressDenyName || item.Name == nsInfo.portGroupIngressDenyName
+	}
+	defaultDenyPGs, err := libovsdbops.FindPortGroupsWithPredicate(oc.nbClient, defaultDenyPGPred)
+	if err != nil {
+		return fmt.Errorf("unable to find default deny port groups %s, %s, err: %v",
+			nsInfo.portGroupEgressDenyName, nsInfo.portGroupIngressDenyName, err)
 	}
```

Comment:
I think it is better to just get the two port groups by name, as it is indexed, than using a predicate. That way we don't have to worry about scale.